geopandas/io/tests/test_file.py

```diff
@@ -687,11 +687,6 @@ def test_infer_zipped_file(engine, nybb_filename):
     gdf = read_file("file+file://" + path, engine=engine)
     assert isinstance(gdf, geopandas.GeoDataFrame)
 
-    # Check that it can add a zip scheme for a path that includes a subpath
-    # within the archive.
-    gdf = read_file(path + "!nybb.shp", engine=engine)
```

Comment:
A `!` isn't intended to mean `/vsizip/`. Only if the URI has a zip scheme.

Reply:
But that is not what we assume or test here. This checks that 
```py
geopandas.read_file("geopandas/tests/data/nybb_16a.zip!nybb.shp")
```
works correctly. That we can infer that the file is zip even if it does not end with `".zip"`.  No part of our code is assuming that `!` == zip. If the change in Fiona caused that this does not work as expected, we need to fix the code, not remove the test.

Reply:
I think there is some confusion here - geopandas is wrapping fiona.ParsedPath.from_uri as part of its logic to automatically infer if something is a zip file, when a scheme isn't explicit in the provided path. This fiona / rasterio bugfix means breaks the (ab)use in geopandas for zip detection.

So I think we could resolve this with something like https://github.com/m-richards/geopandas/pull/12/files#diff-85b3b586d937731f1f0fcc7135cd1a871a2579b45148b66272f3eb02f60f2c35R191-R205 - where geopandas defines the logic it needs directly, and correspondingly adding this test back in. (I didn't want to subsume this PR - really appreciate you getting the ball rolling on this, but I wanted to test this option on CI before suggesting it here)

Reply:
Okay. I have pushed aaab3be adding the test back when engine == pyogrio and added a small to-do note, assuming you will turn your [fiona_1.10_compat](https://github.com/m-richards/geopandas/tree/fiona_1.10_compat) branch to a PR fixing it. Happy to merge this as it is now.

Reply:
Alright, that's a way forward, let's merge this an pick up the zip issue in a follow up.